ferry.go

```diff
@@ -100,23 +100,13 @@ func (f *Ferry) Initialize() (err error) {
 	f.logger.Infof("hello world from %s", VersionString)
 
 	// Connect to the database
-	sourceConfig, err := f.Source.MySQLConfig()
-	if err != nil {
-		f.logger.WithError(err).Error("failed to build config for source database")
-		return err
-	}
-
-	sourceDSN := sourceConfig.FormatDSN()
-	maskedSourceDSN := MaskedDSN(sourceConfig)
-
-	f.logger.WithField("dsn", maskedSourceDSN).Info("connecting to the source database")
-	f.SourceDB, err = sql.Open("mysql", sourceDSN)
+	f.SourceDB, err = f.Source.SqlDB(f.logger.WithField("destination", "source"))
```

Comment:
`destination` is a bit confusing for me since it has a similar meaning to `target`

Reply:
could be `db` or `dbName` or something

Reply:
Yeah I didn't have the best name for it, I'll push a commit changing it to dbname.